src/plugins/saved_objects_management/public/management_section/recent_work.tsx

```diff
@@ -115,51 +126,143 @@ export const RecentWork = (props: { core: CoreStart; workspaceEnabled?: boolean
+        page: 1,
+      };
+      const res = await findObjects(core.http, findOptions);
+      const savedObjects = res.savedObjects
+        .map((obj) => {
+          const findWorkspace = workspaceList.find((workspace) =>
+            obj.workspaces?.includes(workspace.id)
           );
 
           return {
-            ...recentAccessItem,
             ...obj,
-            ...recentAccessItem.meta,
             updatedAt: moment(obj?.updated_at).valueOf(),
+            lastAccessedTime: moment(obj?.updated_at).valueOf(),
             workspaceName: findWorkspace?.name,
+            label: obj.meta.title,
+            link: constructInAppUrl(obj),
           };
+        })
+        .sort(sortBy(sortKeyMap[recentlyUpdated]));
+      setDetailedSavedObjects(savedObjects);
+    } finally {
+      setIsLoading(false);
+    }
+  }, [constructInAppUrl, core.http, currentWorkspace, workspaceList]);
+
+  const getRecentAccessed = useCallback(() => {
+    setIsLoading(true);
+    try {
+      const savedObjects = recentAccessed
+        .filter(
+          (item) =>
+            item.meta?.type &&
+            (!currentWorkspace || (currentWorkspace && item.workspaceId === currentWorkspace.id))
+        )
+        .map((item) => ({
+          type: item.meta?.type || '',
+          id: item.id,
+        }));
+
+      if (savedObjects.length) {
+        bulkGetDetail(savedObjects, core.http).then((res) => {
+          const formatDetailedSavedObjects = res
+            .map((obj) => {
+              const recentAccessItem = recentAccessed.find(
+                (item) => item.id === obj.id
+              ) as ChromeRecentlyAccessedHistoryItem;
+
+              const findWorkspace = workspaceList.find(
+                (workspace) => workspace.id === recentAccessItem.workspaceId
+              );
+
+              const { link, label, workspaceId } = recentAccessItem;
+              const basePath = core.http.basePath;
+
+              const href = formatUrlWithWorkspaceId(link, workspaceId || '', basePath);
+
+              return {
+                ...recentAccessItem,
+                ...obj,
+                ...recentAccessItem.meta,
+                workspaceName: findWorkspace?.name,
+                updatedAt: moment(obj?.updated_at).valueOf(),
+                link: href,
+                label: label || obj.meta.title,
+              };
+            })
+            .sort(sortBy(sortKeyMap[recentlyViewed]));
+          setDetailedSavedObjects(formatDetailedSavedObjects);
         });
-        setDetailedSavedObjects(formatDetailedSavedObjects);
-      });
+      } else {
+        setDetailedSavedObjects([]);
+      }
+    } finally {
+      setIsLoading(false);
     }
-  }, [core.savedObjects.client, recentAccessed, core.http, workspaceList]);
+  }, [core.http, currentWorkspace, recentAccessed, workspaceList]);
+
+  useEffect(() => {
+    // reset to allOption
+    setSelectedType(allOption);
```

Comment:
Q: why reset to allOption is needed? I see `selectedSort` is in deps array of this `useEffect`, it looks like whenever I change the selected type, it resets to `allOption`?

Reply:
when switch between recently viewed and updated, we would like to clear the type filter

![image](https://github.com/user-attachments/assets/5d092d4e-bfa2-44d4-ae50-670f1c12e996)